lib/matplotlib/axes/_axes.py

```diff
@@ -2723,27 +2723,32 @@ def broken_barh(self, xranges, yrange, **kwargs):
 
     @_preprocess_data()
     def stem(self, *args, linefmt=None, markerfmt=None, basefmt=None, bottom=0,
-             label=None, use_line_collection=True):
+             label=None, use_line_collection=True, orientation='horizontal'):
```

Comment:
```suggestion
             label=None, use_line_collection=True, orientation='horizontal'):
```

This is an awkward interpretation of orientation. I claim that the existing and default orientation should be 'vertical'. IMHO orientation should describe the orientation of the stem lines, not the direction of *locs*.

Reply:
I agree. This should also be consistent with `hist`, which is `orientation='horizontal'` for a sideways histogram (independent in y).  And `barh` means the bars are horizontal.

Reply:
Good point, that was the original request, too. I'm not sure at what point this got flipped.